assets/js/components/Dashboard/PermanentConfiguration/Workflows/GlEink/GlEinkWorkflow.tsx

```diff
@@ -32,6 +49,86 @@ const GlEinkWorkflow: ComponentType<WorkflowProps> = ({
     });
   };
 
+  const generateErrorMessage = (errorSet: Set<string>) => {
+    if (errorSet.size === 0) {
+      return "";
+    }
+
+    const capitalizedErrors = Array.from(errorSet).map((error) => {
+      const capitalizedWords = error
+        .split("_")
+        .map(
+          (word: string) =>
+            word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
+        );
+      return capitalizedWords.join(" ");
+    });
+
+    if (capitalizedErrors.length === 1) {
+      return "Correct the following error: " + capitalizedErrors[0] + ".";
+    } else {
+      return (
+        "Correct the following errors: " + capitalizedErrors.join(", ") + "."
+      );
+    }
+  };
+
+  const validateDuplicateScreenIds = (
+    placesAndScreens: PlaceIdsAndNewScreens,
+    duplicateScreenIds: string[] = []
+  ) => {
+    for (const [place_id, screens] of Object.entries(placesAndScreens)) {
+      // Check if screen id is a duplicate
+      screens["new_pending_screens"]?.map((screen, index) => {
+        if (duplicateScreenIds.includes(screen.new_id ?? "")) {
+          validationErrors[place_id][index].isDuplicateScreenId = true;
+        } else {
+          validationErrors[place_id][index].isDuplicateScreenId = false;
+        }
+      });
+    }
+  };
+
+  const validateRequiredFields = (placesAndScreens: PlaceIdsAndNewScreens) => {
+    const fieldsWithErrors = new Set<string>();
+    for (const [place_id, screens] of Object.entries(placesAndScreens)) {
+      const fieldsByScreen = screens["new_pending_screens"]?.map((screen) => {
+        const presentFields = [];
+
+        // Validate that screen id is in correct format
+        if (screen.new_id?.match(/^EIG-\d+$/)) {
```

Comment:
Should we consider the ID field invalid if it is still set to the default of `EIG-`? That is technically a valid ID, but more than likely always be a mistake if it is ever submitted without numbers after the hyphen.

Reply:
Yeah I wondered about that as well. The designs technically do suggest that `EIG-` should be treated as an invalid ID though so I just went with that https://www.figma.com/file/FKWRVrSdoMjZOLorhuykvc/Permanent-Configuration?type=design&node-id=1622-23109&mode=design&t=eKbCkI2xTlb8X1UU-4 let me know if you think we should push back on that though

Reply:
Ahh, I misunderstood your regex here. I thought it was looking for at least `EIG-` but it actually is doing what I was thinking in my comment. My bad, disregard.